builtin/providers/openstack/resource_openstack_compute_instance_v2.go

```diff
@@ -0,0 +1,1011 @@
+	if err != nil {
+		return fmt.Errorf(
+			"Error waiting for instance (%s) to become ready: %s",
+			server.ID, err)
+	}
+	floatingIP := d.Get("floating_ip").(string)
+	if floatingIP != "" {
+		networkingClient, err := config.networkingV2Client(d.Get("region").(string))
+		if err != nil {
+			return fmt.Errorf("Error creating OpenStack compute client: %s", err)
+		}
+
+		allFloatingIPs, err := getFloatingIPs(networkingClient)
+		if err != nil {
+			return fmt.Errorf("Error listing OpenStack floating IPs: %s", err)
+		}
+		err = assignFloatingIP(networkingClient, extractFloatingIPFromIP(allFloatingIPs, floatingIP), server.ID)
+		if err != nil {
+			fmt.Errorf("Error assigning floating IP to OpenStack compute instance: %s", err)
+		}
+	}
+
+	// were volume attachments specified?
+	if v := d.Get("volume"); v != nil {
+		vols := v.(*schema.Set).List()
+		if len(vols) > 0 {
+			if blockClient, err := config.blockStorageV1Client(d.Get("region").(string)); err != nil {
+				return fmt.Errorf("Error creating OpenStack block storage client: %s", err)
+			} else {
+				if err := attachVolumesToInstance(computeClient, blockClient, d.Id(), vols); err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return resourceComputeInstanceV2Read(d, meta)
+}
+
+func resourceComputeInstanceV2Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	computeClient, err := config.computeV2Client(d.Get("region").(string))
+	if err != nil {
+		return fmt.Errorf("Error creating OpenStack compute client: %s", err)
+	}
+
+	server, err := servers.Get(computeClient, d.Id()).Extract()
+	if err != nil {
+		return CheckDeleted(d, err, "server")
+	}
+
+	log.Printf("[DEBUG] Retreived Server %s: %+v", d.Id(), server)
+
+	d.Set("region", d.Get("region").(string))
+	d.Set("name", server.Name)
+	d.Set("access_ip_v4", server.AccessIPv4)
+	d.Set("access_ip_v6", server.AccessIPv6)
+
+	hostv4 := server.AccessIPv4
+	if hostv4 == "" {
+		if publicAddressesRaw, ok := server.Addresses["public"]; ok {
+			publicAddresses := publicAddressesRaw.([]interface{})
+			for _, paRaw := range publicAddresses {
+				pa := paRaw.(map[string]interface{})
+				if pa["version"].(float64) == 4 {
+					hostv4 = pa["addr"].(string)
+					break
+				}
+			}
+		}
+	}
+
+	// If no host found, just get the first IPv4 we find
+	if hostv4 == "" {
+		for _, networkAddresses := range server.Addresses {
+			for _, element := range networkAddresses.([]interface{}) {
+				address := element.(map[string]interface{})
+				if address["version"].(float64) == 4 {
+					hostv4 = address["addr"].(string)
+					break
+				}
+			}
+		}
+	}
+	d.Set("access_ip_v4", hostv4)
+	log.Printf("hostv4: %s", hostv4)
+
+	hostv6 := server.AccessIPv6
+	if hostv6 == "" {
+		if publicAddressesRaw, ok := server.Addresses["public"]; ok {
+			publicAddresses := publicAddressesRaw.([]interface{})
+			for _, paRaw := range publicAddresses {
+				pa := paRaw.(map[string]interface{})
+				if pa["version"].(float64) == 4 {
```

Comment:
Right, I meant the refactor patch I have in queue. It removes the check for a "public" network entirely.